torch_xla/csrc/runtime/distributed_runtime.h

```diff
@@ -0,0 +1,35 @@
+#ifndef XLA_CLIENT_DISTRIBUTED_RUNTIME_H_
+#define XLA_CLIENT_DISTRIBUTED_RUNTIME_H_
+
+#include <memory>
+
+#include "xla/pjrt/distributed/distributed.h"
+
+namespace torch_xla {
+namespace runtime {
+
+class DistributedRuntime {
+ public:
+  static DistributedRuntime& getInstance(int global_rank) {
```

Comment:
Sure, that sounds fine to me! My request is just that we don't require the full list of constructor arguments each time we retrieve the instance.

Reply:
hm, using std::call_once didn't work well so I might as well use the original approach. Btw, any reason for having a separate `Get()` (I guess for getting the DistributedRuntime obj) and a `Initialize(global_rank, master_addr, master_port)` (initializing the DistributedRuntime and returns DistributedRuntime)?

Reply:
In the general use case, we'll be retrieving the initialized distributed runtime multiple times after creation, and it's cumbersome to have to pass in all constructor args in the get call.

Reply:
Is the existing `std::shared_ptr<xla::DistributedRuntimeClient> DistributedRuntime::GetClient() {` what you need?

Reply:
That's right, but since that's an instance method we need to first access the singleton `DistributedRuntime` object. Just to double check my understanding:
- This change introduces DistributedRuntime as a class which contains a singleton object tracking the DistributedRuntimeService and DistributedRuntimeClient
- Right now, the proposed method to access the singleton instance is `getInstance(...)`, which requires all parameters necessary to initialize the distributed runtime service and client

Is that correct? If so, I'm proposing to split the initialization and retrieval so that `Get`-ing the singleton instance doesn't require any parameters. Happy to sync offline if I'm missing anything.

Reply:
ah I see what you means.

Reply:
> Can we make this simply Get() and add another static method for Initialize(global_rank, master_addr, master_port)?

I couldn't find a way to do it because the singleton pattern deletes the copy and assign constructor so I can't save the instance.